### PR TITLE
Initial pass at Task development docs (and others)

### PR DESCRIPTION
**Pull request by aarontp**
I mostly created this for the developer documentation on how to create a new task, but I also pulled in the contribution doc, and started outlining the bigger picture of how Turbinia works (because I wanted to cover some of the details of Tasks and Evidence that I thought would fit better there than in the developer docs).  The how-it-works page isn't complete, but it's a start.

Note that the contributing doc is almost completely taken from the Timesketch project.  I just made a few Turbinia specific tweaks.

### CODE REVIEW

**Comment by berggren** [comment]
I moved this into a doc instead so we can collaborate more easily.

File: docs/how-it-works.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+and handles management of Evidence, Jobs, Tasks and Workers
+
+## Task manager flow
+Tasks are configured to "listen" for specific Evidence types, and if the Task
+Maanger sees a new piece of Evidence (either from a new Turbinia request, or
+because another Task generated a new piece of Evidence), and there is a Task
+that configured to run for that type of Evidence, then the Task Maanger will
+schedule a new Task for it.
+
+## Evidence
```
Comment (berggren) [inline docs/how-it-works.md]: I suggest we put this before the Task manager stuff above seeing we are referencing it there.

File: docs/how-it-works.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
+# How Turbinia Works
+## General
+This page contains some of the details on the internals of how Turbinia works.
+
+## Tasks
+A Task is the smallest discrete unit of schedulable work.  Tasks are sheduled
```
Comment (berggren) [inline docs/how-it-works.md]: s/sheduled/scheduled

File: docs/how-it-works.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+execute it.  After the Task is complete (successfully or not), the Worker will
+return the status, any error logs and results.
+
+## Task Manager
+The Task Manager Acts as a broker between the clients and workers
+and handles management of Evidence, Jobs, Tasks and Workers
+
+## Task manager flow
+Tasks are configured to "listen" for specific Evidence types, and if the Task
+Maanger sees a new piece of Evidence (either from a new Turbinia request, or
```
Comment (berggren) [inline docs/how-it-works.md]: s/Maanger/Manager

File: docs/how-it-works.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+because another Task generated a new piece of Evidence), and there is a Task
+that configured to run for that type of Evidence, then the Task Maanger will
```
Comment (berggren) [inline docs/how-it-works.md]: s/Maanger/Manager

File: docs/how-it-works.md (hunk 2 above)
Comment (aarontp) [reply docs/how-it-works.md]: Done.

File: docs/how-it-works.md (hunk 3 above)
Comment (aarontp) [reply docs/how-it-works.md]: Done.

File: docs/how-it-works.md (hunk 4 above)
Comment (aarontp) [reply docs/how-it-works.md]: Done.

File: docs/how-it-works.md (hunk 1 above)
Comment (aarontp) [reply docs/how-it-works.md]: Done.

**Pull request merged**